### PR TITLE
Remove WebDatasets from Composer

### PR DESCRIPTION
**Pull request by ravi-mosaicml**
Remove webdatasets from Composer, as we have our own streaming dataset format.

### CODE REVIEW

File: composer/datasets/ffcv_utils.py [hunk 1]
```diff
@@ -85,6 +84,7 @@ def write_ffcv_dataset(dataset: Optional[Dataset] = None,
     if dataset:
         writer.from_indexed_dataset(dataset, chunksize=chunk_size)
     elif remote is not None:
+        raise NotImplementedError("FFCV with remote datasets is not implemented")
```
[the remark this answers is not in the document]
Comment (knighton) [reply composer/datasets/ffcv_utils.py:87]: *remove

[the remark this answers is not in the document]
Comment (dskhudia) [reply composer/datasets/ffcv_utils.py:87]: Can we modify this such that it works with YOLO dataset?

[the remark this answers is not in the document]
Comment (dskhudia) [reply composer/datasets/ffcv_utils.py:87]: FFCV will be unusable if we don't do it.

**Comment by dblalock** [comment]
one-off comment: `scripts/create_ffcv_datasets.py` also mentions WebDataset in the argparse help

File: composer/datasets/ffcv_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (abhi-mosaic) [reply composer/datasets/ffcv_utils.py:87]: @dskhudia could you clarify here? If the user has a FFCV dataset on hand, it seems like they will still be able to read it and use in Composer. If the issue is specifically with writing FFCV datasets, maybe we can leave that up to users to implement, and keep some scripts around for our internal use (like in `/scripts/mds/...`).

Also it looks like `write_ffcv_dataset` may be more like `write_ffcv_image_classification_dataset` right? The `image` and `class` keys would not suffice for writing ADE20k. Wonder if it's worth trying to make this bit re-usable, or if it's just for our internal research.

[the remark this answers is not in the document]
Comment (ravi-mosaicml) [reply composer/datasets/ffcv_utils.py:87]: Removed this code, created https://mosaicml.atlassian.net/browse/CO-394 for adding this into the YOLO datasets.

**Comment by ravi-mosaicml** [comment]
> one-off comment: `scripts/create_ffcv_datasets.py` also mentions WebDataset in the argparse help

Thanks! Updated the help to say streaming instead of web datasets. I'm thinking to leave the stubs for `remote` in here, as it will be added back in with the new streaming support? Currently it will raise the not implemented error.

**Review by dskhudia — APPROVED** [review]
FFCV related changes look good to me.

**Review by knighton — APPROVED** [review]
"Let me play you the song of my people" -webdataset

**Comment by knighton** [comment]
https://www.youtube.com/watch?v=RyYhbC0MXlY

**Pull request merged**